### PR TITLE
Create a static weather component which shows the current weather

### PR DESCRIPTION
**Pull request by Jigao-Zeng**
**Issue:**
#15 
User can't get the current weather information, so they can't prepare to face the current weather

**Solution:**
Create an interface showing the current weather 

**Test Method:** 
It shows the current Auckland weather in the browser, I am not sure of other ways to test it, any suggestions? 

**Risk:**
include weather component in the 'App.js'

**Reviewed By:**
[Edit the message as to who reviewed it here]

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,28 +1,55 @@
-import React from "react";
```
Comment (Denijar) [inline client/src/App.js:1]: To prevent conflicts with future PRs, there should not be any changes in the App.js file.

File: client/src/components/Weather/style.css [hunk 2]
```diff
@@ -0,0 +1,36 @@
+body{
```
Comment (Denijar) [inline client/src/components/Weather/style.css:1]: for consistency, can we use module.scss files instead (so rename to style.module.scss)? [CSS Modules information](https://css-tricks.com/css-modules-part-1-need/) This should be very easy for you to convert. Every CSS selector must start with the class selector. You can remove the *body* styling (the only one that doesn't have a class selector) and add background-color: white; to .container. Then, in index.js, you can change all of the className="example" to: className={styles.example}

File: client/src/components/Weather/style.css [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    
+}
+
+
+
+.icon-container{
+    float:left;
+}
+
+.discription{
```
Comment (Denijar) [inline client/src/components/Weather/style.css:33]: nit: description :)

File: client/src/components/Weather/index.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        </div>
+        
+     )
+    
+  }
+
+
+export default Weather;
+
+// If you want to start measuring performance in your app, pass a function
```
Comment (Denijar) [inline client/src/components/Weather/index.js:113]: What are these comments for?

File: client/src/components/Weather/index.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    
+
+    const [icon,setIcon] = useState(undefined);
+    const [main,setMain] = useState(undefined);
+    const [celsius, setCelsius] = useState(undefined);
+    const [temp_max, setMaxTemp] = useState(null);
+    const [temp_min, setMinTemp] = useState(null);
+    const [description, setDescription] = useState("");
+   
+    /* the get_WeatherIcon functions set the weather icon based on the weather condition codes
```
Comment (Denijar) [inline client/src/components/Weather/index.js:32]: nice commenting

**Review by Denijar — CHANGES_REQUESTED** [review]

File: client/src/components/Weather/index.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCloud } from '@fortawesome/free-solid-svg-icons';
+import { faCloudRain } from '@fortawesome/free-solid-svg-icons';
+import { faBolt } from '@fortawesome/free-solid-svg-icons';
+import { faSnowflake } from '@fortawesome/free-solid-svg-icons';
+import { faSun } from '@fortawesome/free-solid-svg-icons';
+import { faSmog } from '@fortawesome/free-solid-svg-icons';
+import './style.css';
+
+const API_KEY = "8e20c3b423d1a8139959af74dcb9cba2";
```
Comment (Chuppay) [inline client/src/components/Weather/index.js:12]: For security reasons, you should be storing the API key in the code itself. You should store it elsewhere like a text file or something that can be read into the code but isn't uploaded onto git so no-one knows your personal API key.

File: client/src/components/Weather/index.js [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    Clouds: <FontAwesomeIcon icon={faCloud} />,
+}
+
+function Weather()  {
+    
+
+    const [icon,setIcon] = useState(undefined);
+    const [main,setMain] = useState(undefined);
+    const [celsius, setCelsius] = useState(undefined);
+    const [temp_max, setMaxTemp] = useState(null);
```
Comment (Chuppay) [inline client/src/components/Weather/index.js:28]: I think it would be best to have the variable and the set variable function have the same names.  You should try and stick to one [tempMax,setTempMax] or [maxTemp,setMaxTemp]

File: client/src/components/Weather/index.js [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    const [temp_min, setMinTemp] = useState(null);
+    const [description, setDescription] = useState("");
+   
+    /* the get_WeatherIcon functions set the weather icon based on the weather condition codes
+       see https://openweathermap.org/weather-conditions for more details
+       cloud drizzle icon needs pro plan, so the cloud rain icon is used instead
+    */
+    const get_WeatherIcon = (rangeId) => {
```
Comment (Chuppay) [inline client/src/components/Weather/index.js:36]: nit-pick: getWeatherIcon

File: client/src/components/Weather/style.css (hunk 2 above)
Comment (Chuppay) [reply client/src/components/Weather/style.css:1]: You will also need to use camelCase because you cannot get the class from the module in index.js if it is kebab-case

**Review by Chuppay — COMMENTED** [review]
Great Work! Just need to fix a few things to follow code convention :)

File: client/src/components/Weather/index.js (hunk 7 above)
Comment (Jigao-Zeng) [reply client/src/components/Weather/index.js:28]: Thank you, I will keep that consistent.

File: client/src/components/Weather/style.css (hunk 2 above)
Comment (Jigao-Zeng) [reply client/src/components/Weather/style.css:1]: 屏幕截图(614)
屏幕截图(615)
@Denijar I follow your advice and make the changes as shown in the image, I think I am doing it wrong cause my css feature is gone, can you see what I am doing wrong here?

Comment (Denijar) [reply client/src/components/Weather/style.css:1]: you need to wrap the `styles.description` in curly braces instead of quote marks. e.g. `className={styles.description}` You also need to have `import styles from `./style.module.scss` at the top of the file

File: client/src/App.js [hunk 9]
```diff
@@ -1,28 +1,55 @@
...
-import logo from "./logo.svg";
-import "./style.css";
+import logo from './logo.svg';
+import './style.css';
+import Weather from './components/Weather'
 
 function App() {
-    return (
```
Comment (dikunong) [inline client/src/App.js:6]: As @Denijar has mentioned in the above comment, App.js should be changed in a separate PR. This PR should only focus on the Weather widget - you should revert changes outside of that widget to avoid affecting other PRs.  I noticed that you resolved the comment but this code is still being deleted at the moment.

File: client/src/components/Weather/index.js [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+          setIcon(weatherIcon.Clouds);
+          break;
+        default:
+          setIcon(weatherIcon.Clouds);
+      }
+    }
+    
+  
+    const toCelsius = (temp) => {
+      let cell = Math.floor(temp - 273.15);
```
Comment (dikunong) [inline client/src/components/Weather/index.js:66]: Magic numbers should be avoided as they can confuse contributors who edit the code later on and aren't familiar with what they mean. You could try defining this number in a variable at the top of the file and then referring to that variable here, perhaps like:

```
const kelvinToCelsiusDiff = 273.15;
```
```
let cell = Math.floor(temp - kelvinToCelsiusDiff);
```

File: client/src/components/Weather/index.js (hunk 4 above)
Comment (dikunong) [inline client/src/components/Weather/index.js:113]: It's best to avoid committing code that is commented out as it isn't providing any functionality and may confuse people who may wonder why this has been left behind. If there's important information here that needs to be referred to for documentation purposes, the wiki is the best place for it :)

**Review by dikunong — CHANGES_REQUESTED** [review]
Awesome work so far, just a few things that could be cleaned up to make it easier for others to understand!

**Review by Denijar — CHANGES_REQUESTED** [review]
You can make a snapshot test for this component. Rename the `test.js` file to `Weather.test.js` then look at [this file](https://github.com/SE701Group2/daily-focus/pull/70/files#diff-61ce5108ae913924a9b204ae655060195bb3a235525114806ab53037fb39259a) (lines 14-18 plus the relevant imports) to see how to write a snapshot test. Run `npm test` and it should automatically generate the snapshot file.

File: client/src/components/Weather/HourCard/HourCard.module.scss [hunk 11]
```diff
@@ -0,0 +1,6 @@
+/*.hourCard{
```
Comment (Denijar) [inline client/src/components/Weather/HourCard/HourCard.module.scss:1]: Leaving commented code in the codebase is confusing. I notice that you are still using the hourCard class styling on line 78 of the HourCard/index.js file. Were you meant to uncomment this code?

File: client/src/components/Weather/HourCard/index.js [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+                setIcon(weatherIcon.Clouds);
+        }
+    };
+
+    const toCelsius = (temp) => {
+        let cell = Math.floor(temp - 273.15);
+        return cell;
+    };
+
+    /*let date = new Date();
```
Comment (Denijar) [inline client/src/components/Weather/HourCard/index.js:58]: Please remove this commented code if it is no longer needed.

File: client/src/components/Weather/HourCard/index.js [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+
+    // Will display time in 10:30:23 format
+    let formattedTime = hours + ":" + minutes.substr(-2) + ":" + seconds.substr(-2);
+
+    useEffect(() => {
+        getWeatherIcon(props.hour.weather[0].id);
+    });
+
+    return (
+        <div className={styles.hourCard}>
```
Comment (Denijar) [inline client/src/components/Weather/HourCard/index.js:78]: As I mentioned in a previous comment, the styling this refers to is currently commented out.

**Review by Denijar — CHANGES_REQUESTED** [review]
This component is getting there! Impressed with the functionality you have managed to achieve. I think your styling needs some work. There is also quite a bit of leftover commented out code in this PR - can that please be removed. Can't wait to see the finished product :)

File: client/package.json [hunk 14]
```diff
@@ -51,7 +52,7 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "lint-staged"
+      "pre-commit": "npx lint-staged"
```
[the remark this answers is not in the document]
Comment (Jigao-Zeng) [reply client/package.json]: I 
屏幕截图(619)
 can't commit my changes in the "gitdesktop" gui, but I can commit in terminal, so I change it back now.

File: client/src/components/Weather/HourCard/HourCard.module.scss (hunk 11 above)
Comment (Jigao-Zeng) [reply client/src/components/Weather/HourCard/HourCard.module.scss]: fix now

File: client/src/components/Weather/HourCard/index.js (hunk 12 above)
Comment (Jigao-Zeng) [reply client/src/components/Weather/HourCard/index.js]: removed now thanks

File: client/src/App.js (hunk 9 above)
Comment (Jigao-Zeng) [reply client/src/App.js]: Ok, It should be the same with everyone's App.js now

**Review by Denijar — CHANGES_REQUESTED** [review]
Getting there, the component is looking much better. There are cases of unused variable in the code which is blocking `npm start` from working - you need to ensure that you can run the component before you commit and push. There is still work to do on the styling. Try not to use volute positioning because it causes stuff like this: 
image

so close!!
